src/data_handling/timeseries.jl

```diff
@@ -48,6 +53,24 @@ function timeseries(notes, property, f, grid)
     return tvec, ts
 end
 
+"""
+    segment_notes(notes, grid)
+Quantizes the positions and durations of `notes` and segments them according to the duration of a grid unit.
+"""
+function segment_notes(notes, grid)
+    tpq = notes.tpq
+    segment_duration = Int(grid[2]*tpq)
+    grid_division = Int(tpq/segment_duration)
+    qnotes = quantize(notes, 0:1//grid_division:1)
+    segmented_notes = Notes()
+    for qn in qnotes
+        for segment_idx in 0:round(Int,qn.duration/segment_duration)-1
+            push!(segmented_notes, Note(qn.pitch, qn.velocity, qn.position + segment_idx*segment_duration, segment_duration))
```

Comment:
Here `Note` should be replaced by the actual note type that the `notes` container uses, so that this works for any kind of notes. (For example the TD50 data don't use `Note` but `MoreVelNote` because they allow for velocities more than 127).

Reply:
I'm not sure I know how to do that.

Reply:
I can do that once you take care of the other comments.